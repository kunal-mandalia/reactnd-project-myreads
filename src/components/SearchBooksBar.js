import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class SearchBooksBar extends Component {
  state = {
    searchText: ''
  }

  handleChange (value) {
    this.setState({ searchText: value })
    value === '' ? this.props.handleClearSearch() : this.props.handleSearch(value)
  }

  render () {
    return (
      <div className="search-books-bar">
        <Link to='/' className='close-search'>Close</Link>
        <div className="search-books-input-wrapper">
          {/* 
            NOTES: The search from BooksAPI is limited to a particular set of search terms.
            You can find these search terms here:
            https://github.com/udacity/reactnd-project-myreads-starter/blob/master/SEARCH_TERMS.md
            
            However, remember that the BooksAPI.search method DOES search by title or author. So, don't worry if
            you don't find a specific author or title. Every search is limited by search terms.
          */}
          <input
            id='search-input'
            type="text"
            placeholder="Search by title or author"
            onChange={(e) => this.handleChange(e.target.value)}
            value={this.state.searchText}
          />
        </div>
      </div>
    )
  }
}

SearchBooksBar.propTypes = {
  handleSearch: PropTypes.func.isRequired,
  handleClearSearch: PropTypes.func.isRequired
}

export default SearchBooksBar
