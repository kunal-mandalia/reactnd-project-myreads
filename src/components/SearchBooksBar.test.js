import SearchBooksBar from './SearchBooksBar'
import renderer from 'react-test-renderer'
import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { mount } from 'enzyme'
import MountOptions from '../../helper/mountOptions'

const mockHandleSearch = jest.fn();
const mockHandleClearSearch = jest.fn();

const props = {
  handleSearch: mockHandleSearch,
  handleClearSearch: mockHandleClearSearch
}

describe('<SearchBooksBar .../>', () => {
  it('should update search query on input', () => {
    const value = 'learning react'
    const wrapper = mount(<SearchBooksBar {...props} />, MountOptions)
    wrapper.find('#search-input').simulate('change', {target: {value}})
    expect(wrapper.state().searchText).toEqual(value)
  })

  it('should match snapshot', () => {
    const tree = renderer.create(<MemoryRouter><SearchBooksBar {...props} /></MemoryRouter>)
    expect(tree).toMatchSnapshot()
  })
})
