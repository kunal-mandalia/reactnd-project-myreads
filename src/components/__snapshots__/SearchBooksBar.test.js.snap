exports[`<SearchBooksBar .../> should match snapshot 1`] = `
<div
  className="search-books-bar">
  <a
    className="close-search"
    href="/"
    onClick={[Function]}>
    Close
  </a>
  <div
    className="search-books-input-wrapper">
    <input
      autoFocus={true}
      id="search-input"
      onChange={[Function]}
      placeholder="Search by title or author"
      type="text"
      value="" />
  </div>
</div>
`;
